### ISSUE TITLE
[feat]: Update Vaul to the latest release

### ISSUE THREAD

**Issue by AmirHmZz** [shadcn-ui/ui#5087]
### Feature description

Vaul has lots of improvements and bug fixes since version 0.9.0. It would be better to upgrade it to the latest version.

### Affected component/components

Drawer

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues and PRs

**Comment by skirsten** [comment]
Yes, please. I just spend a hour banging my head why I can't get the snap points working like at https://vaul.emilkowal.ski/snap-points.
The upgrade fixed it.

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to two years of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you.

**Comment by shadcn** [comment]
No stale. Something went wrong with the stale bot.

**Comment by shadcn** [comment]
Done.
